### PR TITLE
[estimation/sparse/newton] New primal-dual interior-point solver for …

### PR DESCRIPTION
**Pull request by trendelkampschroer**
…reversible MLE problem

### CODE REVIEW

File: msmtools/estimation/sparse/newton/mle_rev.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+
+    if full_output:
+        info = {'z': []}
+        info['z'].append(z)
+        
+    while True:
+        if show_progress:
+            l=z[N+P:N+P+M]
+            s=z[N+P+M:]
+            print "%i %.6e %.6e %.6e %.6e %.6e %s" %(n+1, mu, dual, prim,
```
Comment (marscher) [inline msmtools/estimation/sparse/newton/mle_rev.py]: python3 will not like this (from **future** import print_statement).

File: setup.py [hunk 2]
```diff
@@ -115,6 +115,11 @@ def extensions():
                   include_dirs=[np_inc,
                                 ])
 
+    mle_trev_sparse_newton_module = \
+        Extension('msmtools.estimation.sparse.newton.objective_sparse',
+                  sources=['msmtools/estimation/sparse/newton/objective_sparse.pyx'],
```
Comment (marscher) [inline setup.py]: add include_dirs=[np_inc], to fix compilation.

**Comment by franknoe** [comment]
Thanks a lot!
Do we have any significant tests for reversible P-matrix estimation? I have only found a single test example for this and for the fixed-point iteration each. I think we should give a few test examples, and we should tests for consistency of both estimators in the limit of many iterations.

**Comment by marscher** [comment]
@trendelkampschroer Do you mind checking  the "enable edits from maintainers" box in this PR on the Github website, so I can apply my fixes for your PR? Sorry for taking such a long time - how are you doing by the way? :)

**Comment by trendelkampschroer** [comment]
Gladly, but can't find the button on the site

**Comment by marscher** [comment]
On 24.06.2018 14:19, Benjamin Trendelkamp-Schroer wrote:
> Gladly, but can't find the button on the site
> 
You should find it on the PR page on the bottom right side

**Comment by trendelkampschroer** [comment]
Will try on Desktop later. On mobile right now.

**Comment by trendelkampschroer** [comment]
Ok, done!

**Comment by trendelkampschroer** [comment]
Ä

**Comment by marscher** [comment]
Thank you very much!

**Pull request merged**